### PR TITLE
Updated Set-AzAutomationModule  to use PUT call while updating modules with specific versions

### PR DESCRIPTION
**Pull request by adishiritwick**
## Description
https://github.com/Azure/azure-powershell/issues/12552

Set-AzAutomationModule was not importing updated Module because PATCH call was allowing  updating of tags only. That's why replaced it with PUT call

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by msJinLei** [comment]
@adishiritwick please add change description to https://github.com/Azure/azure-powershell/blob/89c1fb31d7135560522c06f4234bf021bbe28a11/src/Automation/Automation/ChangeLog.md

File: src/Automation/Automation/Common/AutomationPSClient.cs [hunk 1]
```diff
@@ -390,22 +390,23 @@ public Module GetModule(string resourceGroupName, string automationAccountName,
...
+                        ContentHash = null,
+                        Version = (String.IsNullOrWhiteSpace(contentLinkVersion))
                         ? Guid.NewGuid().ToString()
-                        : contentLinkVersion;
-
-                moduleUpdateParameters.Tags = moduleModel.Tags;
-
-                this.automationManagementClient.Module.Update(resourceGroupName, automationAccountName, name,
-                    moduleUpdateParameters);
+                        : contentLinkVersion
```
[the remark this answers is not in the document]
Comment (adishiritwick) [reply src/Automation/Automation/Common/AutomationPSClient.cs]: put call doesn't accept tags as I checked in this repo and even in RP

File: src/Automation/Automation/Common/AutomationPSClient.cs [hunk 2]
```diff
@@ -390,22 +390,23 @@ public Module GetModule(string resourceGroupName, string automationAccountName,
         {
             var moduleModel =
                 this.automationManagementClient.Module.Get(resourceGroupName, automationAccountName, name);
-            if (contentLinkUri != null)
+            if (moduleModel!=null && contentLinkUri != null)
```
[the remark this answers is not in the document]
Comment (adishiritwick) [reply src/Automation/Automation/Common/AutomationPSClient.cs]: build was failing if I made ContentLink Uri is now mandatory. that's why reverted it.
If the module model is null, I am throwing error

File: src/Automation/Automation/Common/AutomationPSClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jagpreetstamber) [reply src/Automation/Automation/Common/AutomationPSClient.cs]: ok

**Comment by dingmeng-xue** [comment]
The title and description of pull request is vague. Please update it.

**Comment by dingmeng-xue** [comment]
@adishiritwick , as Lei mentioned, you need to update changelog. Otherwise, upcoming Az release cannot detect this change.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**